design/FY2019/NFP-HAMT-Enhancements-Pt1.md

```diff
@@ -0,0 +1,307 @@
+## Justification for New Feature ##
+The current HAMT model does not include air leakage in its calculation of the moisture deposition. Air movement as a moisture transport mechanism is typically far more important than vapor diffusion. Air leaks through walls, roofs, and floors can have the most damaging effect on the durability of a house. Uncontrolled airflow through the shell not only carries moisture into framing cavities, causing mold and rot, but it also can account for a huge portion of a home's energy use and can cause indoor air quality problems.
+
+## Overview ##
+The modification adds an internal moisture source in HAMT components due to air flow through the component. The moisture source is attached to a user-selected layer inside the component. Input options to determine the air flow are added, including a user-defined air flow, air flow determined from stack height, air permeance of the component and ventilation overpressure as well as are flow computed with the AirflowNetwork model.
+
+These modifications to EnergyPlus are part of a larger effort to enhance the moisture transport capabilities of EnergyPlus. This NFP describes the first two steps of this process:
+
+Step 1 - Enable air leakage to be applied as a source/sink at any arbitrary point inside the simulation model. Manual inputs of air flow rates is required.
+
+Step 2 - Allow air leakage rates to be controlled by air pressure gradients induced by wind forces, temperature differences and ventilation system.
+
+The necessary inputs will be developed to allow both user specified airflows and airflows calculated using the AirflowNetwork feature. The HAMT model calculations will be modified to account for this additional mode of transport.
+
+## Approach ##
+The HAMT model subdivides a wall into set of cells and computes the movement of moisture through the wall. Equation (3.36) in the Engineering Reference describes the calculation that is done. A source term that adds moisture to a specific cell is added to this equation of the form
+
+```
+S_i A_ij
+```
+
+where
+
+```
+S_i = m_dot(c_src - c_i_sat)
+```
+
+and
+
+`m_dot` - air flow through the envelope per unit area (m^3/(m^2 h))
+
+`c_src` - water vapor concentration in the source volume (kg/m^3)
+
+`c_i_sat` - water vapor saturation concentration for cell `i` (kg/m^3)
+
+## Engineering Reference ##
+The HAMT section of the Engineering Reference will be modified to include and describe the new source term.
+
+## I/O Reference ##
+The I/O Reference will be modified to describe the additional moisture source inputs. A moisture source object is attached to a particular layer in a manner similar to the way that heat sources are specified. Two options are being considered.
+
+### Modify Existing Object ###
+
+The first option modifies the existing internal sources object to include the needed moisture-related parameters. This is the preferred approach as it makes fewer changes to the IDD.
+
+```
+Construction:InternalSources,
```

Comment:
Questions:
1. Is it feasible for HAMT to have more than one moisture source in a given construction?
2. Would more than one moisture source be useful?
3. Does HAMT currently support internal heat sources?
4. Could HAMT support both a moisture source and a heat source in the same construction? (I guess the proposed IDD object says Yes to this.)

Reply:
@mjwitte Thanks for the review, here are some answers:

1. I believe so, yes. I don't think there are any technical complications that would prevent it from being done.
2. Possibly, that's a point of active discussion right now. For situations where moisture accumulates near the surface of a material we might want to be able to have separate sources to account for which direction the air is going.
3. I think so, but I'll need to check to be sure.
4. Same as for number 3. The IDD changes were mainly driven by the possibility of multiple sources, but I'd like to be flexible in any case.